cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -270,51 +269,44 @@ func GetSchedulerProbeAddress(cfg *kubeadmapi.ClusterConfiguration) string {
 	return "127.0.0.1"
 }
 
-// GetEtcdProbeAddress returns the etcd probe address
-func GetEtcdProbeAddress(cfg *kubeadmapi.Etcd) string {
+// GetEtcdProbeURL takes a kubeadm Etcd configuration object and attempts to parse
+// the first URL in the listen-metrics-urls argument, returning an etcd probe hostname,
+// port and scheme
+func GetEtcdProbeURL(cfg *kubeadmapi.Etcd) (string, int, v1.URIScheme) {
+	localhost := "127.0.0.1"
 	if cfg.Local != nil && cfg.Local.ExtraArgs != nil {
-		if arg, exists := cfg.Local.ExtraArgs[etcdListenClientURLsArg]; exists {
-			// Use the first url in the listen-client-urls if multiple url's are specified.
+		if arg, exists := cfg.Local.ExtraArgs[EtcdListenMetricsURLsArg]; exists {
+			// Use the first url in the listen-metrics-urls if multiple URL's are specified.
 			if strings.ContainsAny(arg, ",") {
 				arg = strings.Split(arg, ",")[0]
 			}
 			parsedURL, err := url.Parse(arg)
-			if err != nil || parsedURL.Hostname() == "" {
-				return "127.0.0.1"
+			if err != nil {
+				return localhost, kubeadmconstants.EtcdMetricsPort, v1.URISchemeHTTP
 			}
-			// Return the IP if the URL contains an address instead of a name.
-			if ip := net.ParseIP(parsedURL.Hostname()); ip != nil {
-				// etcdctl doesn't support auto-converting zero addresses into loopback addresses
-				if ip.Equal(net.IPv4zero) {
-					return "127.0.0.1"
-				}
-				if ip.Equal(net.IPv6zero) {
-					return net.IPv6loopback.String()
-				}
-				return ip.String()
+			// Parse scheme
+			scheme := v1.URISchemeHTTP
+			if parsedURL.Scheme == "https" {
+				scheme = v1.URISchemeHTTPS
```

Comment:
I suspect that HTTPS will not work without providing the client certificates.
So if the scheme is HTTPS, the probe may not work correctlly.

Reply:
> So if the scheme is HTTPS, the probe may not work correctlly :(

indeed it does not - like you found out.
i was thinking that the returned status will be fine even (if it's not authorized), but the HTTPS probe just restarts the container.

HTTP on localhost works.

> This can happen only on if a user has specified extra args. In that case the user will have the responsibility to get the rest of the stuff (including the certs) right.

yes, but i suggest we still allow it, because maybe one day the v1.Probe will support certs for its GET operations!